pkgs/dartpad_shared/lib/headers.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+const _loggingOnHeaderName = 'X-Logging-On';
+
+/// A class that represents the custom request headers for DartPad.
+///
+/// Standard serialization is not used, because the header name should have
+/// prefix 'X-'.
+class DartPadRequestHeaders {
+  static final instance = DartPadRequestHeaders();
+
+  /// The default value is true.
+  ///
+  /// The value is set to false for clients where debug asserts are enabled.
+  /// It can be set to false in constructor.
+  late final bool loggingOn;
+
+  DartPadRequestHeaders({bool? loggingOn}) {
```

Comment:
nit: I would rename this to `enableLogging` to match other flags in the DartPad project (enableGenAi, enable-experiments, etc)